stac_generator/raster_data/generator.py

```diff
@@ -29,105 +39,133 @@ def __init__(
+                return bands
+
+            source_df["bands"] = source_df["bands"].apply(parse_bands)
+
+        # Pass the modified DataFrame to the base class
         super().__init__(
             source_df=source_df,
             collection_cfg=collection_cfg,
             catalog_cfg=catalog_cfg,
             href=href,
         )
 
-    def create_item_from_config(self, source_cfg: RasterSourceConfig) -> List[pystac.Item]:
-        logging.info(f"Processing raster from: {source_cfg.location}")
-        
+    def create_item_from_config(self, source_cfg: RasterSourceConfig) -> list[pystac.Item]:
+        logging.info("Processing raster from: %s", source_cfg.location)
+
         try:
             with rasterio.open(source_cfg.location) as src:
                 bounds = src.bounds
                 bbox = [bounds.left, bounds.bottom, bounds.right, bounds.top]
                 transform = src.transform
                 crs = src.crs
-                
-                geometry = {
-                    "type": "Polygon",
-                    "coordinates": [[
-                        [bounds.left, bounds.bottom],
-                        [bounds.left, bounds.top],
-                        [bounds.right, bounds.top],
-                        [bounds.right, bounds.bottom],
-                        [bounds.left, bounds.bottom]
-                    ]]
-                }
-
-                # Create base item
+                shape = [src.height, src.width]
+
+                # Create geometry as Shapely Polygon
+                geometry = Polygon(
+                    [
+                        (bounds.left, bounds.bottom),
+                        (bounds.left, bounds.top),
+                        (bounds.right, bounds.top),
+                        (bounds.right, bounds.bottom),
+                        (bounds.left, bounds.bottom),
+                    ]
+                )
+                geometry_geojson = mapping(geometry)
+
+                if source_cfg.datetime is None:
+                    raise ValueError("source_cfg.datetime cannot be None")
+
+                # Ensure datetime is timezone-aware
+                if source_cfg.datetime.tzinfo is None:
+                    datetime_aware = source_cfg.datetime.replace(tzinfo=UTC)
+                else:
+                    datetime_aware = source_cfg.datetime
+
+                # Create STAC Item
                 item = pystac.Item(
                     id=source_cfg.prefix,
-                    geometry=geometry,
+                    geometry=geometry_geojson,
                     bbox=bbox,
-                    datetime=source_cfg.datetime,
-                    properties={}  # properties will be managed by extensions
+                    datetime=datetime_aware,
+                    properties={
+                        "eo:snow_cover": 0,
+                        "eo:cloud_cover": 0,
+                        "proj:epsg": crs.to_epsg(),
```

Comment:
As I suggested with the vector stac generator, you should compare that the epsg code provided in the config file is the same as the one read from the file.

Reply:
done